instrumentation/apache-dubbo-2.7/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/TracingFilter.java

```diff
@@ -40,9 +40,13 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) {
       return invoker.invoke(invocation);
     }
 
-    boolean isServer = rpcContext.isProviderSide();
-    Instrumenter<DubboRequest, Result> instrumenter =
-        isServer ? serverInstrumenter : clientInstrumenter;
+    Instrumenter<DubboRequest, Result> instrumenter;
+    if (clientInstrumenter != null && serverInstrumenter != null) {
```

Comment:
imo the logic here is hard to follow, need to know when which of the instrumenters is set. I'd use something like https://github.com/steverao/opentelemetry-java-instrumentation/pull/7

Reply:
> imo the logic here is hard to follow, need to know when which of the instrumenters is set. I'd use something like [steverao#7](https://github.com/steverao/opentelemetry-java-instrumentation/pull/7)

Thanks!